src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -20,15 +22,16 @@ internal class Http3Connection : IRequestProcessor
 
         public DynamicTable DynamicTable { get; set; }
 
-        public Http3ControlStream SettingsStream { get; set; }
+        public Http3ControlStream ControlStream { get; set; }
         public Http3ControlStream EncoderStream { get; set; }
         public Http3ControlStream DecoderStream { get; set; }
 
-        private readonly ConcurrentDictionary<long, Http3Stream> _streams = new ConcurrentDictionary<long, Http3Stream>();
+        internal readonly ConcurrentDictionary<long, Http3Stream> _streams = new ConcurrentDictionary<long, Http3Stream>();
 
-        // To be used by GO_AWAY
-        private long _highestOpenedStreamId; // TODO lock to access
-        //private volatile bool _haveSentGoAway;
+        internal long _highestOpenedStreamId; // TODO lock to access
+        internal long _lastStreamProcessed;
+        private volatile bool _haveSentGoAway;
+        internal object _sync = new object();
```

Comment:
Making _streams, _sync and _lastStreamProcessed internal feels bad.  Can we move the implementation Http3ControlStream.OnClientGoAway int Http3Connection to avoid this?